flink-ml-servable-core/src/main/java/org/apache/flink/ml/servable/types/MatrixType.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.servable.types;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * A DataType representing a map where all keys have the same BasicType and all values have the same
+ * BasicType.
+ */
+@PublicEvolving
+public final class MatrixType extends DataType {
```

Comment:
Should we distinguish DataType for SparseMatrix/DenseMatrix and SparseVector/DenseVector?

Reply:
For now, I think it is simpler to keep it as is without distinguishing between SparseMatrix/DenseMatrix. The code that reads the Matrix can just use the public API of Matrix without knowing whether it is SparseMatrix or DenseMatrix. And the code can use `instance of ` if needed.

I expect this to work well since `mleap` uses the same approach and seems to be OK with it. We can change it if needed in the future.

What do you think?

Reply:
Thanks for the explanation. LGTM.